clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -46,7 +46,7 @@
                 try {
                     InetSocketAddress address = new InetSocketAddress(host, port);
                     if (address.isUnresolved())
-                        throw new ConfigException("DNS resolution failed for url in " + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG + ": " + url);
+                        log.error("DNS resolution failed for url in " + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG + ": " + url);
                     addresses.add(address);
```

Comment:
Do you want to add `address` to `addresses` even when `address` is not resolved?